### PR TITLE
Add warning when using an empty regexp in lineinfile

### PR DESCRIPTION
**Pull request by samdoran**
##### SUMMARY

Reverts #41451 and adds a warning instead.

See #29443 for reasoning behind this change.

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME
`lineinfile.py`

##### ANSIBLE VERSION

```
2.7
```

### CODE REVIEW

File: lib/ansible/modules/files/lineinfile.py [hunk 1]
```diff
@@ -480,12 +480,17 @@ def main():
     regexp = params['regexp']
     line = params.get('line', None)
 
+    if regexp == '':
+        module.warn(
+            'The regular expression is empty, which will match every line in the file.'
```
Comment (abadger) [inline lib/ansible/modules/files/lineinfile.py]: I think we need to say "empty string".  We also may want to give an alternative for people who actually do want to match every line in the file.

File: test/integration/targets/lineinfile/tasks/main.yml [hunk 2]
```diff
@@ -717,37 +717,3 @@
   assert:
     that:
       - result.stat.checksum == 'eca8d8ea089d4ea57a3b87d4091599ca8b60dfd2'
```
Comment (abadger) [inline test/integration/targets/lineinfile/tasks/main.yml]: Keeping tests (but updating if the results have changed) seems like a good idea.  Then it's an additional clue that the behaviour was meant to be this way.

**Comment by abadger** [comment]
Also, I think we want a changelog for this?  (And maybe a porting guide?)

File: lib/ansible/modules/files/lineinfile.py (hunk 1 above)
Comment (samdoran) [reply lib/ansible/modules/files/lineinfile.py]: What do you mean by "give an alternative"? This warning could be annoying to someone who does want this capability, and I don't want to encourage folks to disable warnings with noisy error messages.

Comment (abadger) [reply lib/ansible/modules/files/lineinfile.py]: Give an example of a regex to use if they really want to match every line.  (I think "^"  would do it and be efficient).  We could also give an example of unsetting it but that isn't as succinctly written.

Comment (samdoran) [reply lib/ansible/modules/files/lineinfile.py]: Ah, that's perfect! I see what you're saying.

**Review by achinthagunasekara — APPROVED** [review]

**Pull request merged**